### PR TITLE
Drum kit

### PR DESCRIPTION
**Pull request by AlekseyGrimm**
https://youtu.be/VuN8qwZoego

### CODE REVIEW

File: JS-Drum-Kit/app.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+// Play drum by keyboard press
+window.addEventListener('keydown' , playDrum);
```
Comment (y-solovey) [inline JS-Drum-Kit/app.js]: please move adding event listeners to the bottom of file

File: JS-Drum-Kit/app.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+function playDrum(e) {
+    const audio = document.querySelector(`audio[data-key="${e.keyCode}"]`);
+    if(!audio) return;
+    audio.currentTime = 0;
+    audio.play();
+    const key = document.querySelector(`.key[data-key="${e.keyCode}"]`);
```
Comment (y-solovey) [inline JS-Drum-Kit/app.js]: please move const to the top

File: JS-Drum-Kit/app.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    key.classList.add('playing');
+};
+
+// Remove .playing class after transition action over
+const keys = document.querySelectorAll('.key');
+keys.forEach( key => key.addEventListener('transitionend' , removePlayingClass ) );
+function removePlayingClass(e)
+{
+    //if( e.propertyName != 'transform' ) return;
```
Comment (y-solovey) [inline JS-Drum-Kit/app.js]: please enable the checking

File: JS-Drum-Kit/app.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    this.classList.remove('playing');
+}
+
+
+// Play music by mouse click
+keys.forEach( key => key.addEventListener('click' , playByMouse ) );
+function playByMouse(e)
+{
+    const keyCode = this.getAttribute('data-key');
+	const audio = document.querySelector(`audio[data-key="${keyCode}"]`);
```
Comment (y-solovey) [inline JS-Drum-Kit/app.js]: please move repeating code to the function

**Review by y-solovey — CHANGES_REQUESTED** [review]

File: JS-Drum-Kit/app.js (hunk 1 above)
Comment (AlekseyGrimm) [reply JS-Drum-Kit/app.js]: done

File: JS-Drum-Kit/app.js (hunk 2 above)
Comment (AlekseyGrimm) [reply JS-Drum-Kit/app.js]: done

File: JS-Drum-Kit/app.js (hunk 3 above)
Comment (AlekseyGrimm) [reply JS-Drum-Kit/app.js]: done

File: JS-Drum-Kit/app.js (hunk 4 above)
Comment (AlekseyGrimm) [reply JS-Drum-Kit/app.js]: doesn't work like that

File: JS-Drum-Kit/app.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
+const keys = document.querySelectorAll('.key');
+
+// Play drum by keyboard press
+
```
Comment (y-solovey) [inline JS-Drum-Kit/app.js:3]: you can create a function like
```suggestion
function play(keyCode) {
	const audio = document.querySelector(`audio[data-key="${keyCode}"]`);
	if(!audio) return;
	audio.currentTime = 0;
    audio.play();
    const key = document.querySelector(`.key[data-key="${keyCode}"]`);
    key.classList.add('playing');
}
```

File: JS-Drum-Kit/app.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+function playDrum(e) {
```
Comment (y-solovey) [inline JS-Drum-Kit/app.js]: and change the function to
```suggestion
function playDrum(e) {
play(e.keyCode)
}
```

File: JS-Drum-Kit/app.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+// Remove .playing class after transition action over
+keys.forEach( key => key.addEventListener('transitionend' , removePlayingClass ) );
+function removePlayingClass(e){
+    if( e.propertyName != 'transform' ) return;
+    this.classList.remove('playing');
+};
+
+// Play music by mouse click
+keys.forEach( key => key.addEventListener('click' , playByMouse ));
+function playByMouse(e){
```
Comment (y-solovey) [inline JS-Drum-Kit/app.js]: and this function will be like
```suggestion
function playByMouse(e){
const keyCode = this.getAttribute('data-key');
play(keyCode);
}
```

File: JS-Drum-Kit/app.js (hunk 4 above)
Comment (y-solovey) [reply JS-Drum-Kit/app.js]: please see my comments above

**Review by y-solovey — CHANGES_REQUESTED** [review]

File: JS-Drum-Kit/app.js (hunk 7 above)
Comment (AlekseyGrimm) [reply JS-Drum-Kit/app.js]: done

File: JS-Drum-Kit/app.js (hunk 6 above)
Comment (AlekseyGrimm) [reply JS-Drum-Kit/app.js]: done

File: JS-Drum-Kit/app.js (hunk 5 above)
Comment (AlekseyGrimm) [reply JS-Drum-Kit/app.js:3]: done

File: JS-Drum-Kit/app.js (hunk 4 above)
Comment (AlekseyGrimm) [reply JS-Drum-Kit/app.js]: well done

File: JS-Drum-Kit/app.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+	audio.currentTime = 0;
+    audio.play();
+    const key = document.querySelector(`.key[data-key="${keyCode}"]`);
+    key.classList.add('playing');
+};
+
+// Play drum by keyboard press
+function playDrum(e) {
+    play(e.keyCode)
+    audio.play();
```
Comment (y-solovey) [inline JS-Drum-Kit/app.js]: You don't need a call to this function, please pay attention to the browser console

**Review by y-solovey — CHANGES_REQUESTED** [review]

File: .vscode/settings.json [hunk 9]
```diff
@@ -0,0 +1,3 @@
+{
+    "liveServer.settings.port": 5501
```
Comment (y-solovey) [inline .vscode/settings.json]: please remove this file

File: JS-Drum-Kit/app.js (hunk 8 above)
Comment (AlekseyGrimm) [reply JS-Drum-Kit/app.js]: update this code

File: .vscode/settings.json (hunk 9 above)
Comment (AlekseyGrimm) [reply .vscode/settings.json]: done

**Pull request merged**